storage/pipeline/input.go

```diff
@@ -449,6 +449,9 @@ func (m *Sealing) updateInput(ctx context.Context, sp abi.RegisteredSealProof) e
 		if err != nil {
 			return 0, big.Zero(), err
 		}
+		if onChainInfo == nil {
+			return 0, big.Zero(), xerrors.Errorf("sector info for sector %d not found", sn)
+		}
```

Comment:
Instead of fixing this here, I think I would rather fix `GetSector()` in `chain/actors/builtin/miner/vX.go` (these are generated files). I think we should return an error from `GetSector()` if we fail to find it, instead of the `nil, nil` that we're currently returning. It's not idiomatic and is what caused this issue in the first place.

Reply:
This is documented behaviour, we want some way there to indicate “sector not found” that works over jsonrpc - Im pręty sure other parts of the codebase depend on this behaviour

Reply:
Did Aayush get the typed json-rpc errors PR done? There are other parts of our code that aren't checking for `nil, nil`, such as `warmpup.go` that we should also fix if we're going to keep the behaviour the same. We should also throw comments in a couple more places just to make sure as well.